collect_app/src/main/java/org/odk/collect/android/dao/FormsDao.java

```diff
@@ -44,6 +44,14 @@ public Cursor getFormsCursor(String selection, String[] selectionArgs) {
         return getFormsCursor(null, selection, selectionArgs, null);
     }
 
+    public int deleteFormsForFormId(String[] selectionArgs) {
+        String selection = FormsProviderAPI.FormsColumns._ID + " in (";
+        for(int i = 0; i < selectionArgs.length-1; i++)
+            selection += "?, ";
+        selection += "? )";
+        return deleteFormsForFormId(selection, selectionArgs);
+    }
+
```

Comment:
@nribeka, This is what I tried, by deleting in batches of 999. It will both prevent the 999-error and also keep things running fast. Please review it once.